File: backend/core/interactem/core/models/triggers.py
from enum import Enum

from pydantic import BaseModel

from interactem.core.models.canonical import CanonicalOperatorID


class TriggerInvocation(BaseModel):
    canonical_operator_id: CanonicalOperatorID
    trigger: str


class TriggerInvocationRequest(BaseModel):
    trigger: str


class TriggerInvocationResponseStatus(str, Enum):
    OK = "ok"
    ERROR = "error"


class TriggerInvocationResponse(BaseModel):
    status: TriggerInvocationResponseStatus
    message: str | None = None

File: frontend/interactEM/src/types/triggers.ts
import { z } from "zod"
import type { OperatorSpecTrigger } from "../client"

export type { OperatorSpecTrigger }

export const TriggerInvocationResponseSchema = z.object({
  status: z.enum(["ok", "error"]),
  message: z.string().nullable().optional(),
})
export type TriggerInvocationResponse = z.infer<
  typeof TriggerInvocationResponseSchema
>

export const TriggerInvocationRequestSchema = z.object({
  trigger: z.string(),
})
export type TriggerInvocationRequest = z.infer<
  typeof TriggerInvocationRequestSchema
>
